### ISSUE TITLE
Very poor performance when moderately large missions are loaded

### ISSUE THREAD

**Issue by rubenp02** [mavlink/qgroundcontrol#12236]
## Current Behavior
When working with moderately large missions (of maybe around 250+ simple waypoints, probably more than the typical use case but still much less than what is currently supported in the main autopilot firmwares), QGroundControl becomes very unresponsive to the point of being unusable, particularly in the Plan View and in low end devices such as Android GCS. In the latter case it can and does crash at any moment, making it unsuitable for operations where the GCS is critical.

## Expected Behavior
Adequate performance and stability, at least up to 600 waypoints, which is what ArduPilot supports without an external SD card. Or at the very least the performance of the current QGC stable, where this issue isn't nearly as bad. The model size of even a huge mission is very manageable by any system, so this seems to be due to very innefficient rendering or unnecessary view updates.

## Steps to Reproduce:
1. Create and load a mission with a large number of simple waypoints
2. Pan the view, scroll down the waypoint list, etc.
3. The performance will be abysmal at this point. If using an Android GCS, the app will crash within a few minutes.

## System Information
- QGC Version: current master

## Sample mission
[test-250-simple-wp-mission.zip](https://github.com/user-attachments/files/18195955/test-250-simple-wp-mission.zip)

**Comment by HTRamsey** [comment]
Are you able to run the QML Profiler? If not I will do so at some point. I'm not really aware of anything that's changed with any of the mission/plan view stuff so I'll have to look into it further

**Comment by HTRamsey** [comment]
Haven't tried on Android yet but at least on a high end Linux it works fine.
![image](https://github.com/user-attachments/assets/b06cc88f-49b2-4fec-bda8-507f46db4143)

**Comment by DonLakeFlyer** [comment]
Unless there is come specific control that has gone whacky there isn't much different in the Plan view Stable -> Daily.

**Comment by DonLakeFlyer** [comment]
Tried OSX, don't see anything out of the ordinary

**Comment by DonLakeFlyer** [comment]
Windows is working fine for me as well with the example plan from above

**Comment by DonLakeFlyer** [comment]
Herelink seems fine as well. Scrolling the waypoint list up/down is a little sluggish but still workable.

**Comment by DonLakeFlyer** [comment]
Still odd you would see it on multiple OS

**Comment by DonLakeFlyer** [comment]
@rubenp02 Can you try setting 'cacheBuffer' to 0 here: https://github.com/mavlink/qgroundcontrol/blob/master/src/PlanView/PlanView.qml#L755

Run it with the profile as well and provide output like you did above.

**Comment by DonLakeFlyer** [comment]
Can you do a profile with a plan which just has say 10 waypoint in it?

**Comment by DonLakeFlyer** [comment]
My pull will help this a little, but there is still something significantly wrong here.

**Comment by HTRamsey** [comment]
Do you know the specs of the SkyDroid? I just tried on a samsung tab active 4 pro and it seemed alright. Also turned on GPUWatch and profiled on the android device itself and stayed 50fps+

**Comment by DonLakeFlyer** [comment]
I loaded the 250 WP plan on my Herelink and it all worked fine. No perf deg or crashes.

**Comment by DonLakeFlyer** [comment]
Fix it how?

**Comment by DonLakeFlyer** [comment]
I'll take a look at this again

**Comment by DonLakeFlyer** [comment]
@rubenp02 I've tried all sorts of ways to reproduce this and can't see any problems. Can you do a run yourself with QML profiling on Windows:
* Build release QGC with qml debugging/profiling on
* Start Qml profiling, turn it of if QGC boots with it on
* Go to Plan, get ready for 250 plan load by going to File/Open and file dialog ready to click Open
* Turn on Qml profiling
* Once plan loads turn off qml profiling

**Comment by DonLakeFlyer** [comment]
@rubenp02 Have you tried the latest builds with compiled Qml?

**Comment by rubenp02** [comment]
Yeah, seems much, much better. The crashes were also apparently unrelated, I figured out the reason and the fix was recently merged.

This next week I'll rebase our fork on top of everything and test it out for real, to see if it can finally replace stable for us (the only things that were missing were performance and stability). If all goes well I'll close this issue then.

**Comment by DonLakeFlyer** [comment]
@rubenp02 Have you tried things end to end to see if they've improved?

**Comment by rubenp02** [comment]
Ah, sorry, forgot about this.

I had a full week of flights scheduled with a client and wanted to test it there, but it ended up being delayed until the first week of July, so I haven't tested in real world conditions per se yet. In simulated use it all seemed fully on par with stable though, so I'd say this can be closed. But I can hold off until that first week of July if you want a more thorough review.

**Comment by DonLakeFlyer** [comment]
I'm gonna close and can create new Issue with specific cases if needed. Hoping to mark stable within the next couple days.

**Comment by rubenp02** [comment]
Sounds good, thanks.
